frontend-react/src/components/annotate.js

```diff
@@ -3,73 +3,164 @@ import ReactImageAnnotate from "react-image-annotate";
-                    regionClsList={["kiwi", "leaf"]}
-                    enabledTools="create-polygon, create-box"
-                    onExit={output => {
-                      console.log(JSON.stringify(output.images));
-                    }}
-                  />
-
-                </div> 
-              </div>
-          );}
-        else {
-            return (
-                <div className="border">
-                  <div className="imageDisplay">
-                    <form className="selectForm" id="formUpload">
-                        <input type="file" id="file" multiple={true} accept="image/*" className="fileInputButton" onChange={this.fileSelectedHandler} />
-                        <label htmlFor="file" className="fileLabel">
-                            Upload Images
-                        </label>
-                    </form>
-
-                  </div>
-                </div>
-            );
-        }
+  delete(annotateData){
+    //Set index to current displayed image
+    this.setState({
+      index: this.state.index = annotateData.selectedImage
+    });
+    //If delete alert is confirmed
+    if(window.confirm('Are you sure you want to delete this image?')){
+      var result = [];
+      result = result.concat(this.state.data);
+
+      //Remove current image from data images and decrement index
+      result.splice(annotateData.selectedImage,1);
+      this.setState({
+        data: result,
+        index: this.state.index = this.state.index - 1
+      });
+
+      //If there are no more images left in Annotator
+      if (this.state.index < 0){
+        //Set index back to 0
+        this.setState({
+          index: this.state.index = 0
+        });
       }
+    }
+    //Cancel deletion
+    else{
+      return
+    }
+  }
+
+  incrementIndex(imagesIndex){
+    this.setState({
+      index: this.state.index = imagesIndex.length
+    });
+  }
+
+  //Each time the annotator is instantiated, the current index is saved
+  updateIndex(index){
+    this.setState({
+      index: this.state.index=index.selectedImage
+    });
+  }
+
+  saveAnnotation(annotateData){
+    //Save current images and annotations to state data
+    this.setState({
+      data: annotateData.images,
+      saved: true
+    });
+
+
+  }
+
+  //Open file input dialog
+  clickFileInput(){
+
+    var element = document.querySelector('#root > div > form > label')
+    const mouseClickEvents = ['mousedown', 'click', 'mouseup'];
+    mouseClickEvents.forEach(mouseEventType =>
+      element.dispatchEvent(
+        new MouseEvent(mouseEventType, {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+            buttons: 1
+        })
+      ),
+    );
+  }
+
+  render() {
+    let {data,
+          uploaded,
+          index
+        } = this.state;
+
+    //If annotator has images uploaded
+    if(uploaded && data.length > 0){
+      return (
+      <div className="border">
+
+        <form className="selectForm">
```

Comment:
```suggestion
        <form id="selectForm">
```

Are these forms actually needed anyway?